core-backend/sandbox/src/env.rs

```diff
@@ -130,25 +137,48 @@ impl<E: Ext + IntoExtInfo + 'static> Environment<E> for SandboxEnvironment<E> {
             trap: None,
         };
 
-        let instance =
-            Instance::new(binary, &env_builder, &mut runtime).map_err(|e| BackendError {
-                reason: "Unable to instanciate module",
-                description: Some(format!("{:?}", e).into()),
-                gas_amount: runtime.ext.unset().into_gas_amount(),
-            })?;
+        let instance = match Instance::new(binary, &env_builder, &mut runtime) {
+            Ok(inst) => inst,
+            Err(e) => {
+                return Err(BackendError {
+                    reason: "Unable to instantiate module",
+                    description: Some(format!("{:?}", e).into()),
+                    gas_amount: runtime
+                        .ext
+                        .take()
+                        .expect("method called only once with no clones around; qed")
+                        .into_gas_amount(),
```

Comment:
Repetitive code. Could we add some method?

Reply:
I have a code snipped which solves #841, which also simplifies that code. Temporary solution is here: [96638c0](https://github.com/gear-tech/gear/pull/836/commits/96638c0a99a3f5719f8b6b14a22ec136892e5184)